tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -0,0 +1,756 @@
+	ipID := *(ipConfigProp.PublicIPAddress.ID)
+	ipName = extractResourceName(ipID)
+	return nil, ipName
+}
+
+// destroyIP deletes the IP address by taking it's name as argument.
+func (az *AzureProvider) destroyIP(ctx context.Context, ipName string) (err error) {
+	future, err := az.ipClient.Delete(ctx, az.resourceGroupName, ipName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to delete the public IP: %s", ipName))
+		return
+	}
+	var response bool = false
+	var responseStatus int = 200
+	var r autorest.Response
+	for i := 1; i <= 300; i++ {
+		time.Sleep(1 * time.Second)
+		r, err = future.Result(az.ipClient)
+		response = r.IsHTTPStatus(responseStatus)
+		if response {
+			return nil
+		}
+	}
+	return err
+}
+
+// destroyInstance deletes the instance by taking it's name as argument.
+func (az *AzureProvider) destroyInstance(ctx context.Context, vmName string) (err error) {
+	future, err := az.vmClient.Delete(ctx, az.resourceGroupName, vmName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to delete the instance: %s", vmName))
+		return
+	}
+	err = future.WaitForCompletionRef(ctx, az.vmClient.Client)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to delete the instance: %s", vmName))
+		return
+	}
+	return nil
+}
+
+// destroyNIC deletes the nic by taking it's name as argument.
+func (az *AzureProvider) destroyNIC(ctx context.Context, nicName string) (err error) {
+	future, err := az.nicClient.Delete(ctx, az.resourceGroupName, nicName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to delete the nic: %s", nicName))
+		return
+	}
+	var response bool = false
+	var responseStatus int = 200
+	var r autorest.Response
+	for i := 1; i <= 300; i++ {
+		time.Sleep(1 * time.Second)
+		r, err = future.Result(az.nicClient)
+		response = r.IsHTTPStatus(responseStatus)
+		if response {
+			return nil
+		}
+	}
+	return err
+}
+
+// destroyNSG deletes the security groups by taking it's name as argument.
+func (az *AzureProvider) destroyNSG(ctx context.Context, nsgName string) (err error) {
+	future, err := az.nsgClient.Delete(ctx, az.resourceGroupName, nsgName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to delete the sec group: %s", nsgName))
+		return
+	}
+	var response bool = false
+	var responseStatus int = 200
+	var r autorest.Response
+	for i := 1; i <= 300; i++ {
+		time.Sleep(1 * time.Second)
+		r, err = future.Result(az.nsgClient)
+		response = r.IsHTTPStatus(responseStatus)
+		if response {
+			return nil
+		}
+	}
+	return err
+}
+
+// destroyDisk deletes the disk attached with the instance by taking its name
+// as an argument.
+func (az *AzureProvider) destroyDisk(ctx context.Context, vmInfo compute.VirtualMachine) (err error) {
+	vmStorageProfile := *(vmInfo.VirtualMachineProperties.StorageProfile)
+	vmOSdiskProperties := *(vmStorageProfile.OsDisk)
+	diskName := *(vmOSdiskProperties.Name)
+	az.diskClient.PollingDuration = time.Minute * 10
```

Comment:
What does this do?

Reply:
PollingDuration argument sets the maximum time to poll for an operation after which an error will be thrown. But this line shouldn't be there it's a mistake, I didn't observe any difference by adding that argument to call out Delete operation.